enhancements/api-review/add-ImagePolicy-to-config.openshift.io-v1alpha1.md

```diff
@@ -0,0 +1,336 @@
+    ```
+
+    Alternatively, we can enable it node wide:
+
+    ```yaml
+    default-docker:
+      use-sigstore-attachments: true
+    ```
+- merges cluster scoped with namespaced ImagePolicy as a json string and update `status` of each ImagePolicy.
+Once an ImagePolicy object gets created/updated, container runtime config controller will call the syncHandler.
+Once an ImagePolicy object gets deleted, container runtime config controller will delete the machine config and run the syncHandler
+
+#### Notes
+
+ImagePolicy is defined as namespaced CRD. `namespace: openshift-config` is used for cluster policy.
+
+### Risks and Mitigations
+
+Risk: The cluster scoped and namespace policies overwrite
+
+Mitigation: MCO container runtime config controller makes sure namespaced policies for an image cannot override global definitions when merging cluster openshift-config policies and namespaced policies into status field of ImagePolicy instances.   
+
+### Drawbacks
+
+## Design Details
+
+### Open Questions [optional]
+
+### Test Plan
+
+**Note:** *Section not required until targeted at a release.*
+
+MCO container runtime config controller can add unit tests and e2e tests.
+- unit test: verify the policies from ImagePolicy instances merged correctly and the controller writes correct format policy.json
+- e2e test: verify the MCO writes configuration to the correct location
+
+
+### Graduation Criteria
+
+**Note:** *Section not required until targeted at a release.*
+
+Define graduation milestones.
+
+These may be defined in terms of API maturity, or as something else. Initial proposal
+should keep this high-level with a focus on what signals will be looked at to
+determine graduation.
+
+Consider the following in developing the graduation criteria for this
+enhancement:
+
+- Maturity levels
+  - [`alpha`, `beta`, `stable` in upstream Kubernetes][maturity-levels]
+  - `Dev Preview`, `Tech Preview`, `GA` in OpenShift
+- [Deprecation policy][deprecation-policy]
+
+Clearly define what graduation means by either linking to the [API doc definition](https://kubernetes.io/docs/concepts/overview/kubernetes-api/#api-versioning),
+or by redefining what graduation means.
+
+In general, we try to use the same stages (alpha, beta, GA), regardless how the functionality is accessed.
+
+[maturity-levels]: https://git.k8s.io/community/contributors/devel/sig-architecture/api_changes.md#alpha-beta-and-stable-versions
+[deprecation-policy]: https://kubernetes.io/docs/reference/using-api/deprecation-policy/
+
+**If this is a user facing change requiring new or updated documentation in [openshift-docs](https://github.com/openshift/openshift-docs/),
+please be sure to include in the graduation criteria.**
+
+**Examples**: These are generalized examples to consider, in addition
+to the aforementioned [maturity levels][maturity-levels].
+
+#### Dev Preview -> Tech Preview
+
+- Ability to utilize the enhancement end to end
+- End user documentation, relative API stability
```

Comment:
For the API: We could add custom rekor/fulcio instance support at that stage by extending the API.

Reply:
Added the api extension to dev preview -> tech preview. The rest `Graduation Criteria` sections are from the enhancement template. can you take a look?